### PR TITLE
Check tiff format when upload, download file

### PR DESCRIPTION
**Pull request by quanpython**

### CODE REVIEW

File: studio/app/common/routers/files.py [hunk 1]
```diff
@@ -82,6 +104,40 @@ def accept_files(cls, path: str, file_types: List[str]):
...
+
+    tiff_format_file = join_filepath([dirpath, ".image_shape.json"])
+    try:
+        tiff_format_dict = JsonReader.read(tiff_format_file)
+    except FileNotFoundError:
+        tiff_format_dict = {}
+    tiff_format_dict[file_name] = {"shape": shape}
+
+    with open(tiff_format_file, "w") as f:
+        json.dump(tiff_format_dict, f)
```
Comment (ReiHashimoto) [inline studio/app/common/routers/files.py:136]: Pls use `indent=4` to format json.

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]
In addition to the review comment, there is a following issue. Pls fix it.

This feature cannot read nested files' shape.

Screenshot 2023-12-25 at 14 06 01

File: studio/app/common/routers/files.py [hunk 2]
```diff
@@ -82,6 +104,40 @@ def accept_files(cls, path: str, file_types: List[str]):
...
+        return {}
+
+
+def update_image_shape(workspace_id, relative_file_path):
+    dirpath = join_filepath([DIRPATH.INPUT_DIR, workspace_id])
+    filepath = join_filepath([dirpath, relative_file_path])
+
+    try:
+        img = Image.open(filepath)
+        shape = [img.n_frames, *img.size]
```
Comment (ReiHashimoto) [inline studio/app/common/routers/files.py:124]: Could you consider using `shape` attribute of `tifffile`(or `imageio`) module?

We plan to support multi-channel tiff files.
Some multi-channel images would be provided in 4d array.
For example, Thorlabs' microscopes would generate following tifffiles.

https://drive.google.com/file/d/1lUZZ-jfnrZeguR8yEDrPn72uUy8Py12G/view?usp=drive_link

Using `tifffile` module, we can see each tiff files contains array with shape `(600, 2, 512, 512)` (It means (frames, channels, width, height) respectively). 
But using `PIL.Image`, we can get n_frames is 1  and size is `(512, 512)`, so we cannot find channels info.

**Review by ReiHashimoto — CHANGES_REQUESTED** [review]
In addition to the review comment, please resolve conflicts.

File: studio/app/common/routers/files.py (hunk 2 above)
Comment (quanpython) [reply studio/app/common/routers/files.py:124]: Could you explain how to read this type of file in more detail?

I could see 600 single tiff files after extracting it. 
But it seems both `PIL.Image` and `tifffile` cannot read this single tiff file and throws `PIL.UnidentifiedImageError`, `tifffile.tifffile.TiffFileError`.

Comment (ReiHashimoto) [reply studio/app/common/routers/files.py:130]: @quanpython I read the file with following code.
```python
import tifffile

file_path = "/Users/rei_hashimoto/Downloads/thorlabs/timelapse600/ChanA_001_001_001_001.tif"
img = tifffile.imread(file_path)
print(img.shape)  # (600, 2, 512, 512)
```

Comment (quanpython) [reply studio/app/common/routers/files.py]: Thank you, I implemented it.

**Review by ReiHashimoto — APPROVED** [review]

**Pull request merged**